src/core/core.helpers.js

```diff
@@ -185,11 +185,13 @@ module.exports = function() {
 		if (!helpers.isFinite(x)) {
 			return;
 		}
-		var e = 1;
+		x = Math.abs(x);
 		var p = 0;
-		while (Math.round(x * e) / e !== x) {
-			e *= 10;
-			p++;
+		if (x > 0) {
+			while (x < 0.99 || !helpers.almostWhole(x, 1e-6)) {
```

Comment:
This however fails with large numbers, for example (current version works with these):
```js
		expect(helpers.decimalPlaces(12345678.1234)).toBe(4); // this PR results to 9
		expect(helpers.decimalPlaces(1234567890.1234567)).toBe(7); // this PR results to 6
```

Reply:
Those are good tests and I agree the result is not acceptable. I will review...

Reply:
@costerwi just checking in - are you still planning to update this PR?